src/snowflake/connector/cursor.py

```diff
@@ -121,6 +122,24 @@
 ASYNC_RETRY_PATTERN = [1, 1, 2, 3, 4, 8, 10]
 
 
+# The USE_NANOARROW_CONVERTER is introduced to control which arrow converter to use.
+# default value is None means it will be decided by the session parameter PYTHON_CONNECTOR_USE_NANOARROW whether
+# to use nanoarrow converter or vendored arrow converter, in which case, PYTHON_CONNECTOR_USE_NANOARROW defaults to True
+# if USE_NANOARROW_CONVERTER set to True to False locally,
+# then the local setting has higher priority than the session parameter.
+# or it can be set via the environment variable USE_NANOARROW_CONVERTER, Only string values `true`, `1` will
+# be taken as True, all other values will be taken as false
```

Comment:
Should we rework these comments? Right now it is pretty confusing. I think we should start from the highest priority option, and show how each option impacts the final decision

Reply:
Agreed, and in particular, I don't understand what this means:

> if USE_NANOARROW_CONVERTER set to True to False locally